redwood/src/decryption.rs

```diff
@@ -58,20 +64,25 @@ impl<'a> DecryptionHelper for Helper<'a> {
             .key()
             .clone();
 
-        // Decrypt the secret key with the specified passphrase.
-        let mut pair = key.decrypt_secret(self.passphrase)?.into_keypair()?;
+        for pkesk in pkesks {
+            // Note: this check won't work for messages encrypted with --throw-keyids,
+            // but we don't generate any messages that use it.
+            if pkesk.recipient() == &KeyID::from(key.fingerprint()) {
+                // Decrypt the secret key with the specified passphrase.
+                let mut pair = key
+                    .clone()
+                    .decrypt_secret(self.passphrase)?
+                    .into_keypair()?;
+                pkesk
+                    .decrypt(&mut pair, sym_algo)
+                    .map(|(algo, session_key)| decrypt(algo, &session_key));
+                // Return the fingerprint of the key we decrypted with, this is used in
+                // conjunction with the intended recipient subpacket (see Section 5.2.3.29
+                // of RFC 4880bis) to prevent Surreptitious Forwarding.
```

Comment:
Nit:

```suggestion
                // conjunction with the intended recipient subpacket (see "Intended Recipient Fingerprint"
                // in RFC 4880bis) to prevent Surreptitious Forwarding.
```

Just because RFC 4880bis isn't published yet, so the enumerated reference is still drifting about.  (I've seen all of 5.2.3.{29,30,36} just while trying to pin it down!)  I think it's more useful to know _what_ to look for anyway.

Reply:
Done, thanks. This was copied out of Sequoia's docs IIRC, I'll find the link so you can also suggest this upstream :)

Reply:
Oh, Sequoia has a link so it's not ambiguous: https://gitlab.com/sequoia-pgp/sequoia/-/blob/main/openpgp/src/parse/stream.rs?ref_type=heads#L2084 - up to you on submitting it upstream then. (I think having a link for our inline docs would be overkill especially now that we're using the section header name)

Reply:
Thanks!  I agree that the upstream reference to a specific version is stable and so fine as is.